src/main/java/org/motechproject/mots/service/InchargeService.java

```diff
@@ -45,10 +72,120 @@ public Incharge findByFacilityId(UUID id) {
    */
   @PreAuthorize(RoleNames.HAS_INCHARGE_READ_ROLE)
   public Page<Incharge> searchIncharges(String firstName, String secondName, String otherName,
-      String phoneNumber, String email, String facilityName, Pageable pageable)
+      String phoneNumber, String email, String facilityName, Boolean selected, Pageable pageable)
       throws IllegalArgumentException {
 
     return inchargeRepository.searchIncharges(firstName, secondName, otherName,
-        phoneNumber, email, facilityName, pageable);
+        phoneNumber, email, facilityName, selected, pageable);
+  }
+
+  /**.
+   * Processes CSV file which contains Incharge list and returns list of errors
+   * @param inchargeCsvFile CSV file with Incharge list
+   * @return map with row numbers as keys and errors as values.
+   * @throws IOException in case of file issues
+   */
+  public Map<Integer, String> processInchageCsv(MultipartFile inchargeCsvFile) throws IOException {
+    ICsvMapReader csvMapReader;
+    csvMapReader = new CsvMapReader(new InputStreamReader(inchargeCsvFile.getInputStream()),
+        CsvPreference.STANDARD_PREFERENCE);
+
+    final String[] header = csvMapReader.getHeader(true);
+    final CellProcessor[] processors = getProcessors();
+
+    Map<String, Object> csvRow;
+    Set<String> phoneNumberSet = new HashSet<>();
+    Map<Integer, String> errorMap = new HashMap<>();
+
+    while ((csvRow = csvMapReader.read(header, processors)) != null) {
+      LOGGER.debug(String.format("lineNo=%s, rowNo=%s, chw=%s", csvMapReader.getLineNumber(),
+          csvMapReader.getRowNumber(), csvRow));
+
+      String facilityId = Objects.toString(csvRow.get("FACILITY_ID"), null);
+
+      Optional<Facility> existingFacility = facilityRepository.findByFacilityId(facilityId);
+
+      if (!existingFacility.isPresent()) {
+        errorMap.put(csvMapReader.getLineNumber(), "Facility with this Facility Id does not exist");
+        continue;
+      }
+
+      String phoneNumber = Objects.toString(csvRow.get("PHU in-charge number"), null);
+
+      if (phoneNumberSet.contains(phoneNumber)) {
+        errorMap.put(csvMapReader.getLineNumber(), "Phone number is duplicated in CSV");
+        continue;
+      }
+
+      if (phoneNumber != null) {
+        phoneNumberSet.add(phoneNumber);
+      }
+
+      String name = Objects.toString(csvRow.get("PHU in-charge name"), null);
+
+      if (StringUtils.isBlank(name)) {
+        errorMap.put(csvMapReader.getLineNumber(), "Incharge name is empty");
+        continue;
+      }
+
+      String[] nameParts = name.split("([. ])");
+      List<String> names = Arrays.stream(nameParts).map(part ->
+          part.length() == 1 ? part + "." : part).collect(Collectors.toList());
+
+      if (names.size() < 2) {
```

Comment:
Please use final variable for example "final int twoNames = 2" to make code more readable.